### PR TITLE
Product listing Home page

### PR DESCRIPTION
**Pull request by kaushiknishchay**
Home Page
- Top 3 Products shown from each category

Components Created

- ProductRow  shows a list of Products in a row and auto wraps to new line
- ProductItem Single product component

### CODE REVIEW

File: packages/CB-serverless-frontend/src/base_components/index.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import styled from 'styled-components';
+
+String.prototype.toProperCase = function () {
```
Comment (prasanna1211) [inline packages/CB-serverless-frontend/src/base_components/index.js]: Can you comment over here for the use of this function ?

File: packages/CB-serverless-frontend/src/components/ProductHome.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    this.state = {
+      catData: null
+    };
+  }
+
+
+  componentDidMount() {
+    axios.get('http://localhost:3000/groceries')
+      .then(res => {
+        this.setState({
```
Comment (prasanna1211) [inline packages/CB-serverless-frontend/src/components/ProductHome.js]: Try to use set state as function

File: packages/CB-serverless-frontend/src/components/ProductHome.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      catData: null
+    };
+  }
+
+
+  componentDidMount() {
+    axios.get('http://localhost:3000/groceries')
```
Comment (prasanna1211) [inline packages/CB-serverless-frontend/src/components/ProductHome.js]: Always catch for promise errors

File: packages/CB-serverless-frontend/src/components/ProductHome.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        this.setState({
+          catData: res.data
+        });
+      });
+  }
+
+
+  render() {
+    const { catData } = this.state;
+    console.log('catData', this.state.catData);
```
Comment (prasanna1211) [inline packages/CB-serverless-frontend/src/components/ProductHome.js]: Remove logs

File: packages/CB-serverless-frontend/src/components/ProductHome.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    return (
+      <React.Fragment>
+        <Wrapper>
+          {catData && catData.map(obj => {
```
Comment (prasanna1211) [inline packages/CB-serverless-frontend/src/components/ProductHome.js]: Use lodash for map or any other associated functionalities.

File: packages/CB-serverless-frontend/src/components/ProductHome.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+            const title = obj.category.toProperCase();
+            const items = obj.groceries;
+            return (<ProductRow title={title} key={title} items={items} />);
```
Comment (prasanna1211) [inline packages/CB-serverless-frontend/src/components/ProductHome.js]: Components with more than 2 props, change to line by line.

File: packages/CB-serverless-frontend/src/components/ProductItem.js [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+    left: 0px;
+    width: calc(140% + 10px);
+    -webkit-transform: translate(-16px,0%);
+    -ms-transform: translate(-16px,0%);
+    transform: translate(-12%,40%) rotate(40deg);
+    text-align: center;
+`;
+
+
+class ProductItem extends Component {
```
Comment (prasanna1211) [inline packages/CB-serverless-frontend/src/components/ProductItem.js]: Write it as functional component if no state is present.

File: packages/CB-serverless-frontend/src/components/ProductItem.js [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+  render() {
+    const { name, price, url, isSoldOut } = this.props;
+    return (
+      <ItemWrap
+        containerStyle={{
+          margin: '0 auto',
+        }}>
+        {
+          isSoldOut &&
```
Comment (prasanna1211) [inline packages/CB-serverless-frontend/src/components/ProductItem.js]: When components go more than 2 in a render write it as sub functions which renders each of them.

File: packages/CB-serverless-frontend/src/components/ProductItem.js (hunk 7 above)
Comment (kaushiknishchay) [reply packages/CB-serverless-frontend/src/components/ProductItem.js]: changes done, cant be made functional as a new method besides render() is present

**Pull request merged**